File: problem10.py
import math

# this method is very slow - improve using a prime sieve
def is_prime(n):
    for j in range(2, math.sqrt(n)+1):
        if n%j == 0:
            return False
    return True

s = 2 # add 2 to start, since looping over odd #s
for i in range(3, 2000000, 2):
    if is_prime(i):
        s += i

print s

File: go/problem10.go
package main

import (
    "fmt"
    "math"
)

func is_prime(i int64) bool {
    var j int64 = 0;
    if i < 2 {
        return false
    } else if i == 2 {
        return true
    }
    for j = 3; j <= sqrt(i); j+=2 {
        if i % j == 0 {
            return false
        }
    }
    return true
}

func sqrt(i int64) int64 {
    return int64(math.Sqrt(float64(i)))
}

func main() {
    var i, sum int64
    for i, sum = 3, 2; i < 2000000; i+=2 {
        if is_prime(i) {
            sum += i
        }
    }
    fmt.Printf("%v\n", sum)
}
